### PR TITLE
ci: Add script and CI job to force std::size_t over size_t

### PR DESCRIPTION
**Pull request by paulgessinger**

### CODE REVIEW

File: CI/check_size_t.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+import os
+import argparse
+from fnmatch import fnmatch
+import re
+import sys
+
+ex = re.compile(r"((?<!std::)size_t)")
```
Comment (paulgessinger) [inline CI/check_size_t.py:10]:
```suggestion
ex = re.compile(r"(\b(?<!std::)size_t)\b")
```

File: CI/check_size_t.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        lines[i] = line
+        if n_subs > 0:
+            changed_lines.append((i, oline))
+
+    if fix and len(changed_lines) > 0:
+        file.write_text("\n".join(lines) + "\n")
+
+    return changed_lines
+    #  for m in ex.finditer(content):
+    #  print(f"{file}:{m.start()}: {m.group(1)}")
```
Comment (paulgessinger) [inline CI/check_size_t.py:71]:
```suggestion
```

**Review by andiwand — DISMISSED** [review]
nice :+1:

File: CI/check_size_t.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    args = p.parse_args()
+
+    # walk over all files
+    exit = 0
+    for root, _, files in os.walk("."):
+        root = Path(root)
+        for filename in files:
+            # get the full path of the file
+            filepath = root / filename
+            if filepath.suffix not in (".hpp", ".cpp", ".ipp"):
```
Comment (AJPfleger) [inline CI/check_size_t.py:32]:
```suggestion
            if filepath.suffix not in (".hpp", ".h", ".cpp", ".ipp", ".c"):
```
We also have some `.h` and `.c` files using `size_t`. Or should I jsut move the files to the correct extensions?

**Review by AJPfleger — APPROVED** [review]
I looked through the first 100 files, seems to work.

**Review by AJPfleger — DISMISSED** [review]

**Comment by AJPfleger** [comment]
I just noticed, that we also have cuda files `.cu` and `.cuh` containing `size_t`.

**Comment by paulgessinger** [comment]
@AJPfleger Ok, added now.

**Review by AJPfleger — APPROVED** [review]

File: CI/check_size_t.py (hunk 3 above)
Comment (benjaminhuth) [reply CI/check_size_t.py]: Hmm don't `.c` and `.h` files indicate C files, where `std::size_t` shouldn't be used?

Comment (AJPfleger) [reply CI/check_size_t.py]: In theory yes, but we seem to also have cpp-code inside those files.
I will move them in the next days/week to the "correct" extension :)

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [bd74d01c9da40e3ea08226d03f63c7e3bf665d47]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6487936)!

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995049) |
|:red_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995048) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995047) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995046) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995045) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995044) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995043) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995042) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995041) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995040) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995039) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995038) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995037) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995036) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995035) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995034) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995033) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33995032) |

**Pull request merged**